.github/workflows/pr-labeling-automation.yml

```diff
@@ -0,0 +1,18 @@
+name: 'Pull Request Labeler'
+on:
+    workflow_dispatch:
+    pull_request_target:
+        types: [opened, closed, reopened, edited, ready_for_review, synchronize]
+
+jobs:
+    auto_label_pull_request:
+        permissions:
+            contents: read
+            pull-requests: write
+            issues: write
```

Comment:
Does this action also label the attached issues? If not, I am not sure if this is required. The docs say that the `issues: write` permission is only necessary "if the action needs to create labels that do not already exist in the repository." Not sure if that's intended here. But if not, let's try removing this to avoid giving permissions unnecessarily.